src/test/scala/org/broadinstitute/dsde/firecloud/integrationtest/SearchResultValidation.scala

```diff
@@ -46,6 +49,20 @@ trait SearchResultValidation {
     Await.result(searchDAO.suggestionsFromAll(criteria.copy(size=100), Seq.empty[String]), dur)
   }
 
+  /**
+    * Mimics a 3rd party searching against an ElasticSearch instance using a research purpose filter
+    * from us. Our SearchDAO is used to create and execute the search request, but the research
+    * purpose is fetched identically to how a 3rd party would.
+    */
+  def searchWithResearchPurposeQuery(researchPurpose: ResearchPurpose): SearchResponse = {
```

Comment:
This is what all of the new test cases in `ResearchPurposeSearchSpec` use to pretend to be a 3rd party using the research purpose query endpoint.